azure_functions_devops_build/yaml/yaml_manager.py

```diff
@@ -66,7 +62,24 @@ def _generate_yaml(self, dependencies, vmImage, language_str, platform_str, pack
         return outputText
 
     def _requires_extensions(self):
-        return True if path.exists('extensions.csproj') else False
+        return path.exists('extensions.csproj')
+
+    def _requires_pip(self):
+        return path.exists('requirements.txt')
+
+    def _requires_npm(self):
+        return path.exists('package.json')
+
+    def _requires_npm_build(self):
+        if path.exists('package.json'):
+            with open('package.json', 'r') as f:
+                json_object = json.load(f)
+                return bool(json_object.get('scripts', {}).get('build'))
+            return False
+        return False
+
+    def _requires_mvn(self):
```

Comment:
Do we need to add a Maven requirement here if we are disabling Java support?

Reply:
Reverted Maven change.